scripts/menu/server/sv_menu.lua

```diff
@@ -0,0 +1,405 @@
+}
+
+local function syncServerCtx()
+  local oneSyncConvar = GetConvar('onesync', 'off')
+  if oneSyncConvar == ('on' or 'legacy') then
+    ServerCtxObj.oneSync.type = oneSyncConvar
+    ServerCtxObj.status = true
+  elseif oneSyncConvar == 'off' then
+    ServerCtxObj.oneSyncStatus = false
+  end
+
+  -- Default '' in fxServer
+  local svProjectName = GetConvar('sv_projectname', '')
+  if svProjectName ~= '' then
+    ServerCtxObj.projectName = svProjectName
+  end
+
+  -- Default 30 in fxServer
+  local svMaxClients = GetConvarInt('sv_maxclients', 30)
+  ServerCtxObj.maxClients = svMaxClients
+
+  local txAdminLocale = GetConvar('txAdmin-locale', 'en')
+  ServerCtxObj.locale = txAdminLocale
+
+  debugPrint('Server CTX assigned to GlobalState, CTX:')
+  debugPrint(json.encode(ServerCtxObj))
+  GlobalState.txAdminServerCtx = ServerCtxObj
+end
+
+-- Everytime the txAdmin convars are changed this event will fire
+-- Therefore, lets update global state with that.
+AddEventHandler('txAdmin:events:configChanged', function()
+  debugPrint('configChanged event triggered, syncing GlobalState')
+  syncServerCtx()
+end)
+
+CreateThread(function()
+  -- If we don't wait for a tick there is some race condition that
+  -- sometimes prevents debugPrint lmao
+  Wait(0)
+  syncServerCtx()
+end)
+
+RegisterServerEvent('txAdmin:menu:getServerCtx', function()
+  local src = source
+  TriggerClientEvent('txAdmin:menu:sendServerCtx', src, serverCtxObj)
+end)
+--
+-- [[ End ServerCtxObj ]]
+--
+
+
+RegisterServerEvent('txAdmin:menu:checkAccess', function()
+  local src = source
+  local canAccess = not (adminPermissions[src] == nil)
+  debugPrint((canAccess and "^2" or "^1") .. GetPlayerName(src) ..
+               " does " .. (canAccess and "" or "NOT ") .. "have menu permission.")
+  TriggerClientEvent('txAdmin:menu:setAccessible', src, canAccess)
+end)
+
+RegisterServerEvent('txAdmin:menu:playerModeChanged', function(mode)
+  local src = source
+  if mode ~= 'godmode' and mode ~= 'noclip' and mode ~= 'none' then
+    debugPrint("Invalid player mode requested by " .. GetPlayerName(src) .. " (mode: " .. (mode or 'nil'))
+    return
+  end
+  
+  local allow = PlayerHasTxPermission(src, 'players.playermode')
+  TriggerEvent("txaLogger:menuEvent", src, "playerModeChanged", allow, mode)
+  if allow then
+    TriggerClientEvent('txAdmin:menu:playerModeChanged', src, mode)
+  end
+end)
+
+RegisterServerEvent('txAdmin:menu:healMyself', function()
+  local src = source
+  local allow = PlayerHasTxPermission(src, 'players.heal')
+  TriggerEvent("txaLogger:menuEvent", src, "healSelf", allow)
+  if allow then
+    TriggerClientEvent('txAdmin:menu:healed', src)
+  end
+end)
+
+RegisterServerEvent('txAdmin:menu:healPlayer', function(id)
+  local src = source
+  local parseId = type(id) == 'string' and tonumber(id) or id
+  if not parseId then
+    return
+  end
+  local allow = PlayerHasTxPermission(src, 'players.heal')
+  local playerName = GetPlayerName(id) or "unknown"
```

Comment:
s/your admin/anyone/ as now you're passing this unchecked value to a native no matter if the user has access lol